### DISCUSSION TITLE
How to Create Hydrogel with Both Closed and Interconnected Pores Using Neper?

### DISCUSSION THREAD

**Discussion by riddraw** [neperfepx/neper#926] [Q&A] [answered]
Hello everyone,

I am trying to create a porous hydrogel structure containing both closed pores (isolated) and interconnected pores (pores connected to other pores and the surface). My idea is to use Neper to generate tessellated cells that are elongated along the y-axis, followed by meshing only the surfaces that are not close to or parallel with the XY plane.

Here is my current approach:

Step 1: Create the tessellation
I used the following command to generate elongated cells:

`neper -T -n 100 -domain 'cube(1,1,1)' -morpho 'voronoi,aspratio(1,1,2)' -periodicity all -ori random -o tessellation`

Step 2: Mesh only surfaces with cell orientation theta > 5 degrees and theta < 175 degrees relative to the XY plane

`neper -M tessellation.tess -meshpoly "theta>=5 && theta<=175"`

`neper -V tessellation.msh -print img`

I got the following result:
![img_teta nonxy](https://github.com/user-attachments/assets/de333bd9-ed05-47dd-831b-55cacf6e5a21)

I am not sure if this result matches my goal of generating both isolated and interconnected pores. Could you please help me understand if I am on the right track or if there's a better way to achieve this structure using Neper?

Any advice or suggestions would be greatly appreciated!

Thank you!

**Comment by rquey** [comment, marked as answer]
I'm not sure what this means:
> Step 2: Mesh only surfaces with cell orientation theta > 5 degrees and theta < 175 degrees relative to the XY plane

and the command
```
$ neper -M tessellation.tess -meshpoly "theta>=5 && theta<=175"
```
probably does not do what you think. First, you are using `-meshpoly`, and second, `theta` is not defined for polys (and would otherwise represent a crystal disorientation angle).
